src/python/pants/core/goals/test.py

```diff
@@ -228,6 +236,12 @@ def register_options(cls, register) -> None:
             default=False,
             help="Force the tests to run, even if they could be satisfied from cache.",
         )
+        register(
+            "--show-output",
+            type=ShowOutput,
+            default=ShowOutput.FAILED,
+            help="Show stdout/stderr for these tests.",
+        )
```

Comment:
Encouraged—but not enforced—to add a property for this so that MyPy understands the code. See below for some examples.